google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/SpannerPoolTest.java

```diff
@@ -460,14 +469,29 @@ public void testSpannerPoolKeyEquality() {
             .setUri("cloudspanner:/projects/p/instances/i/databases/d")
             .setCredentials(NoCredentials.getInstance())
             .build();
+    ConnectionOptions options4 =
+        ConnectionOptions.newBuilder()
+            .setUri("cloudspanner:/projects/p/instances/i/databases/d?routeToLeader=true")
+            .setCredentials(NoCredentials.getInstance())
+            .build();
+    ConnectionOptions options5 =
+        ConnectionOptions.newBuilder()
+            .setUri("cloudspanner:/projects/p/instances/i/databases/d?routeToLeader=false")
+            .setCredentials(NoCredentials.getInstance())
+            .build();
 
     SpannerPoolKey key1 = SpannerPoolKey.of(options1);
     SpannerPoolKey key2 = SpannerPoolKey.of(options2);
     SpannerPoolKey key3 = SpannerPoolKey.of(options3);
+    SpannerPoolKey key4 = SpannerPoolKey.of(options4);
+    SpannerPoolKey key5 = SpannerPoolKey.of(options5);
 
     assertNotEquals(key1, key2);
     assertEquals(key2, key3);
     assertNotEquals(key1, key3);
     assertNotEquals(key1, new Object());
+    assertEquals(key3, key4);
+    assertNotEquals(key4, key5);
+    assertEquals(key2, key4);
```

Comment:
From a readability point of view, its a little hard to understand why these assertions work. Do you think adding some comments would help? Let's discuss offline. 

I understand the key4 and key5 won't be equal since LAR boolean property is part of the equals() method but I didn't get how this is equal to key3 and key2.

Reply:
Will add comments in code. key3 will be equal to key4 because not passing in `routeToLeader` in Connection URI is equivalent to passing it as `true`, as routeToLeader is `true` by default.

Reply:
Done. Added comments while creating ConnectionOptions.